pallets/briefs/src/lib.rs

```diff
@@ -315,6 +316,22 @@ pub mod pallet {
             Self::deposit_event(Event::<T>::BriefEvolution(brief_id));
             Ok(())
         }
+
+        /// Extrinsic to cancel a brief
+        #[pallet::call_index(5)]
+        #[pallet::weight(<T as Config>::WeightInfo::cancel_brief())]
+        pub fn cancel_brief(origin: OriginFor<T>, brief_id: BriefHash) -> DispatchResult {
+            let who = ensure_signed(origin)?;
+            let brief = Briefs::<T>::get(brief_id).ok_or(Error::<T>::BriefNotFound)?;
+
+            ensure!(who != brief.applicant, Error::<T>::NotAuthorised);
+
+            BriefContributions::<T>::remove(brief_id);
+            Briefs::<T>::remove(brief_id);
+
+            Self::deposit_event(Event::<T>::BriefCanceled(brief_id));
```

Comment:
Also reserved funds for the brief are not unreserved 
This certainly needs doing